### ISSUE TITLE
A warning at sign-up that usernames can't be changed later on

### ISSUE THREAD

**Issue by Cassolotl** [tootsuite/mastodon#5647]
See title - since usernames can't currently be changed, I think it should (hopefully) be fairly easy to warn people at sign-up that usernames can't be changed. It's a question that comes up semi-frequently from new *and* established members, so I think it could help with relatively little effort.

Related to #177.

* * * *

- [ ] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by matt-auckland** [comment]
I'm thinking we could add a label beneath the username input, much like how we display errors for the inputs. I'm not sure if there is an existing 'warning' class for this sort of thing so I just chose to use orange for the illustration. Warning text is also a placeholder. Text could be revealed on field focus (i.e user clicks into the field).

![image](https://user-images.githubusercontent.com/12672050/32704537-ae85da56-c86b-11e7-867a-87bb7e8a30e9.png)   
(ignore the icons displayed in the fields those are from Lastpass)

**Comment by Cassolotl** [comment]
Oh this is good!

It could say "this username has already been taken" in red if it's already been taken, and if the username is valid it can then switch to saying "your username is permanent and cannot be changed later" in orange? That could help make it less busy - only one urgent message at a time.

**Comment by matt-auckland** [comment]
That'd be good but, there is a case where you can potentially miss out on the message (for a second time, you'd always see it on the first screen). Say if you got to that screen because the username was taken, then entered a new one and clicked sign up, you'd be signed up without seeing the message (a second time).    
Not sure if that matters or not though. Before I write any code I'd like to wait for input from someone else on the design.

**Comment by nightpool** [comment]
@matt-auckland that design looks fine to me, I'm not too worried about missing it the second time.

I agree that ideally the text should be revealed on field focus. I'm worried about adding more static page JS—we try to keep that as small as possible. can this be done with the standard rails helper JS?

**Comment by nightpool** [comment]
if not then whatever

**Comment by matt-auckland** [comment]
@nightpool No idea to be honest, never used Ruby before 😅 
Need to do some reading and research

**Comment by ccoenen** [comment]
account migration in Mastodon 3+ is basically also rename, right? In that case I'd close this.

**Comment by Cassolotl** [comment]
@ccoenen No, because you can't take your posts with you, as far as I know? If I wanted to keep my account and all its posts, faves, bookmarks, boosts, settings, stats, date joined, etc - keep everything the same, but just change the username, that's not currently possible and this issue should stay open.

**Comment by Cassolotl** [comment]
Just checked on mastodon.social - there is no indication on the registration page that usernames cannot be changed later.
